cargo build --no-default-features is broken in MacOS

Our nginx patch calls `cargo build --no-default-features` in the Makefile and build fails in MacOS environment.

I think we have a problem building dynamic lib with boringssl. If I remove `cdylib` from Cargo.toml as follows it works again:

```
diff --git a/Cargo.toml b/Cargo.toml
index 0e46b0d..d33e860 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -72,7 +72,8 @@ debug = true
 debug = true

 [lib]
-crate-type = ["lib", "staticlib", "cdylib"]
+#crate-type = ["lib", "staticlib", "cdylib"]
+crate-type = ["lib", "staticlib"]

 [[bench]]
 name = "benches"
```

Build failure log:

```$ cargo build --no-default-features
   Compiling quiche v0.1.0 (/Users/junho/work/quiche)
error: linking with `cc` failed: exit code: 1
  |
  = note: "cc" "-m64" "-L" "/Users/junho/.rustup/toolchains/stable-x86_64-apple-darwin/lib/rustlib/x86_64-apple-darwin/lib" "/Users/junho/work/quiche/target/debug/deps/quiche.10c760ei2ioajhwh.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.10zlzabgrcm09jtl.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.12majzlazeb5vpt5.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.13a4vur55z4rlen9.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.13yw22benvu43kww.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.15cbthe7lg1bktdx.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.16454s9pcothl737.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.19q16pb3s8p3q0cc.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.1blds5p2pflwpbti.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.1bra8uy5fwxnss8c.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.1d4yg794mls3gk05.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.1deicq6mv8cpiu4k.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.1dmxcidv8lf5etfh.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.1f5n989449hs9pqs.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.1g5ay3f9v4kqmkxk.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.1hn4vlex4ev8rex8.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.1i7es21656nqhww4.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.1jjpglgfxsepuwnj.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.1likv8nv16qroohm.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.1n4pvsq990zh9m4r.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.1nmfljbgxs3eg17r.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.1nwt7wan286fjhss.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.1o68uvumdbirtx3u.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.1pqxcze4908773ez.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.1qgo62mc249f829t.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.1s8wtv7cvxgydqm4.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.1s9215aq23ebxdfm.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.1vpm8f3cw2624gl.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.1vvbfxe2mw9oo7ck.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.1wgh4p7nu5dtnamc.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.1wuqox1wum9sp99z.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.1ysu89hd7pqoo4vd.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.2021gsabi68yrpgp.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.20bbbfrl9foapnaq.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.20o4tshmnhfzep4n.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.22963i6l65zk5miu.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.22sl92jkug0t4kb6.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.2341rwmiwzf8o5yf.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.26k52ql178kouubp.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.273dhxks8hxptft2.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.2c5twaio6fnrj9j.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.2d33bmipd7nrtjlq.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.2d9exg4kjy4ktwtu.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.2en1afpg5kaheagz.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.2ghvae1lj1osn9b2.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.2j1pofedjic984ro.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.2lusi4gfsm561s8m.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.2o5cr30qdg5q0z6z.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.2ojiusrv5nr0fukr.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.2pnx0tsecnli4kq8.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.2r0bedezwzxjptuz.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.2szoq9d4d0q23lae.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.2u5hvwp3c8bx6csd.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.2u7ripz4t4afa9g6.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.2vprjvgih720uvg9.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.2xg15r58g2d0ubrq.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.2z1tmf7qxk7usz9l.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.2zfj5lfp0epwh04x.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.30c6jgtif63ykk5b.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.31z83gxl8k0f0cdv.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.3440ldo053mqmcsm.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.35ghrq2xjkp4bzs6.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.37h8hioya0xpbcd3.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.38vlc36jpjbzzx2d.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.394fy3kc4j1oqc85.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.39s7zjgg4o4zw6ik.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.3c3otdef1eedui4n.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.3e4ywdg1dsq71561.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.3f5w4j9imitpm3b6.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.3ijkx6w87v35gzmv.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.3j1kg9f9uosqh464.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.3jwtpd9lbt5cnvj0.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.3k0evqrh374u33f0.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.3kd71yubss8mszyh.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.3n1vlad7g11g0lfv.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.3nr97x2x0ank0u2y.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.3nyqe8fayebbz787.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.3ph54uxqizxetutv.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.3uhte7eleod740y.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.3xqmaou6cev882k2.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.3zetjg1vsxmljlhk.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.42r4j9v6f0mfe3u3.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.435vajodm4340uiw.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.46hovu48je4bunf1.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.46pi1h4k1z0swnmg.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.49e0lxya69bzymw0.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.4bwm8lncfqynyqey.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.4ecjdh745zeac2cn.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.4j27esps3ccd1kzl.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.4k1agf5e8sitpbuo.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.4kvmrl1j11p4jwjx.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.4kxrd82uutarouyb.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.4nt7zz1f1zbmlvuu.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.4ohfs01i2xbz3b87.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.4rh12kf9d73iv7y1.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.4ry31hjpxjytvsjo.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.4sqldu6rfksdslfk.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.4svi299iorckmw4u.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.4td4iey1v06jq3dg.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.4wuu24zazwwqdwqs.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.4x6q1ussxgd2duvn.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.4x6tiyp032397t40.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.4xxahrbyok0kgflz.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.50zwd16f84cjgcdy.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.51o24sg2ny365znx.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.53ustchskfuawdcn.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.53vdxzfkhztgngok.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.554qiod8rqtonxfz.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.580a7jgst4dfbgza.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.59ctcma6y4ff3vcd.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.5ci3mzpd458sv26q.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.5d51qeo0t76sgovp.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.5fa5qu8zpg0gfzeo.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.5fngudv7f4o6o9t2.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.68fpkdcu8fu9kk2.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.70pe4p17as5ytfl.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.bbmgdd95m34u23y.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.d9jw5tqv3pyflys.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.gvjge80fgoz12to.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.h0vapw77y1x8ufn.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.i0yoxs3z54c5de7.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.j70msspdahr4fqd.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.jij323ia38ymvl0.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.ksf9wrj20k1eunf.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.l4gya13vvxw6f67.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.lw461m7o0oimw0m.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.mayze6esx3tp03y.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.nkqysf1le93uced.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.osqwt9jcnon2cdi.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.q6nkwxntsyow58k.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.qkmta51s89c4zz7.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.qravuxcp4il7gnl.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.rjgzzho5c7pzdfg.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.s2idyxphmf2y2lq.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.t8alxbtjmaocpn5.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.vtbmy4sjzxhcayp.rcgu.o" "/Users/junho/work/quiche/target/debug/deps/quiche.yekfngz785ec66o.rcgu.o" "-o" "/Users/junho/work/quiche/target/debug/deps/libquiche.dylib" "-Wl,-exported_symbols_list,/var/folders/k0/5dm_jwls2z5bb3f_6f_y9jzh0000gn/T/rustc6rDRLD/list" "/Users/junho/work/quiche/target/debug/deps/quiche.56zjsaqhcw1qzc7r.rcgu.o" "-Wl,-dead_strip" "-nodefaultlibs" "-L" "/Users/junho/work/quiche/target/debug/deps" "-L" "/Users/junho/work/quiche/target/debug/build/ring-0ca7ef6d39df7761/out" "-L" "/Users/junho/.rustup/toolchains/stable-x86_64-apple-darwin/lib/rustlib/x86_64-apple-darwin/lib" "/Users/junho/work/quiche/target/debug/deps/liblazy_static-29b10d75700e38d7.rlib" "/Users/junho/work/quiche/target/debug/deps/liblibc-b6a19ca045e00691.rlib" "/Users/junho/work/quiche/target/debug/deps/libring-27d1c13a84b2c148.rlib" "/Users/junho/work/quiche/target/debug/deps/libspin-c5ab1321ce79447d.rlib" "/Users/junho/work/quiche/target/debug/deps/libuntrusted-5d17a911df4fab10.rlib" "/Users/junho/work/quiche/target/debug/deps/liblog-4a953ebd063e1c66.rlib" "/Users/junho/work/quiche/target/debug/deps/libcfg_if-8e035499bffc3aee.rlib" "/Users/junho/.rustup/toolchains/stable-x86_64-apple-darwin/lib/rustlib/x86_64-apple-darwin/lib/libstd-c912eb7850fbecec.rlib" "/Users/junho/.rustup/toolchains/stable-x86_64-apple-darwin/lib/rustlib/x86_64-apple-darwin/lib/libpanic_unwind-1b3312d8fefaf74e.rlib" "/Users/junho/.rustup/toolchains/stable-x86_64-apple-darwin/lib/rustlib/x86_64-apple-darwin/lib/libbacktrace-fd4028660702ce49.rlib" "/Users/junho/.rustup/toolchains/stable-x86_64-apple-darwin/lib/rustlib/x86_64-apple-darwin/lib/libbacktrace_sys-ae9cc847e15164de.rlib" "/Users/junho/.rustup/toolchains/stable-x86_64-apple-darwin/lib/rustlib/x86_64-apple-darwin/lib/librustc_demangle-e76733b3a72b0f17.rlib" "/Users/junho/.rustup/toolchains/stable-x86_64-apple-darwin/lib/rustlib/x86_64-apple-darwin/lib/libhashbrown-f805efcb06ec7300.rlib" "/Users/junho/.rustup/toolchains/stable-x86_64-apple-darwin/lib/rustlib/x86_64-apple-darwin/lib/librustc_std_workspace_alloc-5b52b7d5eaabfff6.rlib" "/Users/junho/.rustup/toolchains/stable-x86_64-apple-darwin/lib/rustlib/x86_64-apple-darwin/lib/libunwind-4ed6dd6cb0feb09f.rlib" "/Users/junho/.rustup/toolchains/stable-x86_64-apple-darwin/lib/rustlib/x86_64-apple-darwin/lib/libcfg_if-dded45c4727acb1f.rlib" "/Users/junho/.rustup/toolchains/stable-x86_64-apple-darwin/lib/rustlib/x86_64-apple-darwin/lib/liblibc-3b8415dad7bbd863.rlib" "/Users/junho/.rustup/toolchains/stable-x86_64-apple-darwin/lib/rustlib/x86_64-apple-darwin/lib/liballoc-86bfd3683a35bfec.rlib" "/Users/junho/.rustup/toolchains/stable-x86_64-apple-darwin/lib/rustlib/x86_64-apple-darwin/lib/librustc_std_workspace_core-af5aff04ef816462.rlib" "/Users/junho/.rustup/toolchains/stable-x86_64-apple-darwin/lib/rustlib/x86_64-apple-darwin/lib/libcore-62d44948c379a9ba.rlib" "/Users/junho/.rustup/toolchains/stable-x86_64-apple-darwin/lib/rustlib/x86_64-apple-darwin/lib/libcompiler_builtins-4c0e14a54ecf951d.rlib" "-framework" "Security" "-lSystem" "-lresolv" "-lc" "-lm" "-dynamiclib" "-Wl,-dylib"
  = note: Undefined symbols for architecture x86_64:
            "_SSL_set_ex_data", referenced from:
                quiche::tls::Handshake::set_ex_data::h3d9f912100285f2c in quiche.vtbmy4sjzxhcayp.rcgu.o
            "_SSL_clear", referenced from:
                quiche::tls::Handshake::clear::h3ee9e6c0b94f8f21 in quiche.3c3otdef1eedui4n.rcgu.o
            "_SSL_CIPHER_get_id", referenced from:
                quiche::tls::get_cipher_from_ptr::h070f7a4c707c17fe in quiche.3c3otdef1eedui4n.rcgu.o
            "_SSL_in_early_data", referenced from:
                quiche::tls::Handshake::is_in_early_data::hbeeb401a2c74581e in quiche.3c3otdef1eedui4n.rcgu.o
            "_ERR_peek_error", referenced from:
                quiche::tls::log_ssl_error::haac72759a9eaa273 in quiche.3c3otdef1eedui4n.rcgu.o
            "_SSL_session_reused", referenced from:
                quiche::tls::Handshake::is_resumed::he4a2fe656bf1ee5e in quiche.3c3otdef1eedui4n.rcgu.o
            "_SSL_get_curve_id", referenced from:
                quiche::tls::Handshake::curve::hb595975205ecd780 in quiche.3c3otdef1eedui4n.rcgu.o
            "_SSL_do_handshake", referenced from:
                quiche::tls::Handshake::do_handshake::he4c0ebccf0b42d10 in quiche.3c3otdef1eedui4n.rcgu.o
            "_SSL_get0_alpn_selected", referenced from:
                quiche::tls::Handshake::alpn_protocol::h728ef5e0f03b7f21 in quiche.3c3otdef1eedui4n.rcgu.o
            "_SSL_set_quic_transport_params", referenced from:
                quiche::tls::Handshake::set_quic_transport_params::h6b500e1eff0e8ce9 in quiche.3c3otdef1eedui4n.rcgu.o
            "_RAND_bytes", referenced from:
                quiche::rand::rand_bytes::hadd0a7aabb2080f4 in quiche.2d9exg4kjy4ktwtu.rcgu.o
            "_SSL_get0_param", referenced from:
                quiche::tls::Handshake::set_host_name::h5ff26d68fb2c1715 in quiche.3c3otdef1eedui4n.rcgu.o
            "_ERR_error_string_n", referenced from:
                quiche::tls::log_ssl_error::haac72759a9eaa273 in quiche.3c3otdef1eedui4n.rcgu.o
            "_SSL_get_peer_signature_algorithm", referenced from:
                quiche::tls::Handshake::sigalg::he75ffcee1dbe2a05 in quiche.3c3otdef1eedui4n.rcgu.o
            "_X509_VERIFY_PARAM_set1_host", referenced from:
                quiche::tls::Handshake::set_host_name::h5ff26d68fb2c1715 in quiche.3c3otdef1eedui4n.rcgu.o
            "_SSL_free", referenced from:
                _$LT$quiche..tls..Handshake$u20$as$u20$core..ops..drop..Drop$GT$::drop::haa332dee8f6763ae in quiche.3c3otdef1eedui4n.rcgu.o
            "_SSL_CTX_new", referenced from:
                quiche::tls::Context::new::h3b593a968fcb8f7e in quiche.3c3otdef1eedui4n.rcgu.o
            "_SSL_set_max_proto_version", referenced from:
                quiche::tls::Handshake::set_max_proto_version::he433a6cc31013842 in quiche.3c3otdef1eedui4n.rcgu.o
            "_SSL_set_quiet_shutdown", referenced from:
                quiche::tls::Handshake::set_quiet_shutdown::h0d205b0aee5d3bd7 in quiche.3c3otdef1eedui4n.rcgu.o
            "_SSL_get_curve_name", referenced from:
                quiche::tls::Handshake::curve::hb595975205ecd780 in quiche.3c3otdef1eedui4n.rcgu.o
            "_SSL_get_peer_quic_transport_params", referenced from:
                quiche::tls::Handshake::quic_transport_params::h9440d7b839a2e718 in quiche.3c3otdef1eedui4n.rcgu.o
            "_SSL_CTX_set_alpn_protos", referenced from:
                quiche::tls::Context::set_alpn::h95b21d4312ab2504 in quiche.3c3otdef1eedui4n.rcgu.o
            "_SSL_set_quic_method", referenced from:
                quiche::tls::Handshake::set_quic_method::h51272d716e871c20 in quiche.3c3otdef1eedui4n.rcgu.o
            "_SSL_set_connect_state", referenced from:
                quiche::tls::Handshake::set_state::hf8e3f8c1555f2045 in quiche.3c3otdef1eedui4n.rcgu.o
            "_SSL_CTX_use_PrivateKey_file", referenced from:
                quiche::tls::Context::use_privkey_file::h62e3cf173d18021b in quiche.3c3otdef1eedui4n.rcgu.o
            "_SSL_CTX_set_early_data_enabled", referenced from:
                quiche::tls::Context::set_early_data_enabled::h90a74c6685083ae5 in quiche.3c3otdef1eedui4n.rcgu.o
            "_SSL_CTX_set_alpn_select_cb", referenced from:
                quiche::tls::Context::set_alpn::h95b21d4312ab2504 in quiche.3c3otdef1eedui4n.rcgu.o
            "_SSL_get_ex_data", referenced from:
                quiche::tls::get_ex_data_from_ptr::h5b7045f64778180f in quiche.vtbmy4sjzxhcayp.rcgu.o
            "_SSL_get_ex_new_index", referenced from:
                core::ops::function::FnOnce::call_once::h1236ad4a3e01d80a in quiche.51o24sg2ny365znx.rcgu.o
            "_SSL_set_min_proto_version", referenced from:
                quiche::tls::Handshake::set_min_proto_version::h08d384ffb78a76c1 in quiche.3c3otdef1eedui4n.rcgu.o
            "_SSL_CTX_free", referenced from:
                _$LT$quiche..tls..Context$u20$as$u20$core..ops..drop..Drop$GT$::drop::h93ff93426cfec08e in quiche.3c3otdef1eedui4n.rcgu.o
            "_SSL_CTX_set_keylog_callback", referenced from:
                quiche::tls::Context::enable_keylog::h4d21e0e07a20a35e in quiche.3c3otdef1eedui4n.rcgu.o
            "_SSL_CTX_set_verify", referenced from:
                quiche::tls::Context::set_verify::hd0dc8e088ba85ab2 in quiche.3c3otdef1eedui4n.rcgu.o
            "_SSL_get_current_cipher", referenced from:
                quiche::tls::get_cipher_from_ptr::h070f7a4c707c17fe in quiche.3c3otdef1eedui4n.rcgu.o
            "_SSL_get_error", referenced from:
                quiche::tls::Handshake::get_error::h8ef14ac9a7c1d166 in quiche.3c3otdef1eedui4n.rcgu.o
            "_SSL_quic_write_level", referenced from:
                quiche::tls::Handshake::write_level::h4fa468876d85db84 in quiche.3c3otdef1eedui4n.rcgu.o
            "_SSL_provide_quic_data", referenced from:
                quiche::tls::Handshake::provide_data::h12114a92ab9fdc63 in quiche.3c3otdef1eedui4n.rcgu.o
            "_SSL_CTX_set_default_verify_paths", referenced from:
                quiche::tls::Context::load_ca_certs::h9fcb909b9510b6d6 in quiche.3c3otdef1eedui4n.rcgu.o
            "_TLS_method", referenced from:
                quiche::tls::Context::new::h3b593a968fcb8f7e in quiche.3c3otdef1eedui4n.rcgu.o
            "_SSL_CTX_use_certificate_chain_file", referenced from:
                quiche::tls::Context::use_certificate_chain_file::h35705855a5755389 in quiche.3c3otdef1eedui4n.rcgu.o
            "_SSL_new", referenced from:
                quiche::tls::Context::new_handshake::h57bc97b93099fd42 in quiche.3c3otdef1eedui4n.rcgu.o
            "_SSL_get_signature_algorithm_name", referenced from:
                quiche::tls::Handshake::sigalg::he75ffcee1dbe2a05 in quiche.3c3otdef1eedui4n.rcgu.o
            "_SSL_set_tlsext_host_name", referenced from:
                quiche::tls::Handshake::set_host_name::h5ff26d68fb2c1715 in quiche.3c3otdef1eedui4n.rcgu.o
            "_SSL_set_accept_state", referenced from:
                quiche::tls::Handshake::set_state::hf8e3f8c1555f2045 in quiche.3c3otdef1eedui4n.rcgu.o
          ld: symbol(s) not found for architecture x86_64
          clang: error: linker command failed with exit code 1 (use -v to see invocation)


error: aborting due to previous error

error: Could not compile `quiche`.

To learn more, run the command again with --verbose.```

--- a/src/build.rs
+++ b/src/build.rs
@@ -180,6 +180,11 @@ fn main() {
         println!("cargo:rustc-link-lib=static=ssl");
     }
 
+    // MacOS: Allow cdylib to link with undefined symbols
+    if cfg!(target_os = "macos") {
+        println!("cargo:rustc-cdylib-link-arg=-Wl,-undefined,dynamic_lookup");
+    }
+
     if cfg!(feature = "pkg-config-meta") {
         write_pkg_config();
     }

